### PR TITLE
tests: Only re-run unit-tests if not initially running as root

### LINKED ISSUE #496: Only re-run unit-tests if not initially running as root
Improve `go-test.sh` so that the unit-tests are only run once if initially running as `root`. This is required since it is not clear which user `root` could switch to to run the tests as a non-priv user.

(The intent is for the tests to be run as a non-`root` user normally and that user will then elevates privs to `root` via `sudo` to re-run the tests).

### PR DESCRIPTION
**Pull request by jodh-intel**
Don't run the unit-tests twice if the user that invokes the tests is
already root since both runs would be running as `root`.

Also, ensure the HTML report generated by test_html_coverage() is
readable by all users.

Fixes #496.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: .ci/go-test.sh [hunk 1]
```diff
@@ -27,24 +29,36 @@ function run_as_user
...
 
 	echo "mode: atomic" > "$cov_file"
 
 	for pkg in $test_packages; do
 
-		# Run the unit-tests *twice* (since some must run as root and
-		# others must run as non-root), combining the resulting test
-		# coverage files.
-		for user in non-root root; do
+		if [ $(id -u) -eq 0 ]
```
Comment (devimc) [inline .ci/go-test.sh]: I think you should move this ```if ... else ``` condition before ```for pkg in $test_packages; do```

Comment (jodh-intel) [reply .ci/go-test.sh]: good catch! moving...

Comment (jodh-intel) [reply .ci/go-test.sh]: Branch updated.

**Comment by devimc** [comment]
lgtm

**Pull request merged**